v1/format/format.go

```diff
@@ -1207,10 +1207,18 @@ func (w *writer) writeTermParens(parens bool, term *ast.Term, comments []*ast.Co
 	case ast.String:
 		if term.Location.Text[0] == '`' {
 			// To preserve raw strings, we need to output the original text,
-			// not what x.String() would give us.
 			w.write(string(term.Location.Text))
 		} else {
-			w.write(x.String())
+			// x.String() cannot be used by default because it can change the input string "\u0000" to "\x00"
+			// term.Location.Text could contain more than just the string, extract the first string it can find
+			raw := string(term.Location.Text)
```

Comment:
A drawback of defaulting to the original raw string is that we're changing the behavior of the formatter in maybe unexpected ways. 
E.g. before this change, the statement `a := "\u0061"` would have been formatted to `a := "a"` which is both semantically consistent and easier to read.

Reply:
I'd think the formatter changing the strings at all is more unexpected. If a user went through the trouble to write out "\u0061" I'd assume it was intentional and they'd want to keep it like that? I'd prefer it be a linter warning saying "unicode is hard too read, consider changing it to `a`" then a silent change.

Of course this has been around for a while so users could be relying on this behavior. An alternative solution we could consider, instead post-processing the string returned from `strconv.Quote` and replace any `\x00` with `\u0000`. Maybe extend it in the future to handle the other control characters as well. Then it keeps the same behavior in case someone relies on it?

Reply:
Agreed on not changing string contents. My [most important project](https://github.com/anderseknert/colorized) has always been unformattable because of this! 😆

Reply:
I agree that not rewriting strings is preferable 👍. I just wanted to highlight that this is a behavioral change.
I'd be surprised if anyone actually relied upon this. Maybe if you're generating code and relying on the formatter to clean it up .. 🤔. But I'm not sure we should cater to that 😄.